File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.java
class Solution {
    public int minIncrementForUnique(int[] nums) {
        HashSet<Integer> hs = new HashSet<Integer>();
        Arrays.sort(nums);
        int result = 0;
        
        hs.add(nums[0]);
        int maxValueAdded = nums[0];
        
        for(int i = 1; i < nums.length; i++) {
            if(!hs.contains(nums[i])) {
                hs.add(nums[i]);
                maxValueAdded = nums[i];
            } else {
                hs.add(maxValueAdded + 1);
                maxValueAdded++;
                result += maxValueAdded - nums[i];
            }
        }
        
        return result;
    }
}

File: 0945-minimum-increment-to-make-array-unique/0945-minimum-increment-to-make-array-unique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minIncrementForUnique = function(nums) {
    const mySet = new Set();
    nums.sort((a,b) => a - b);
    let result = 0;
    
    mySet.add(nums[0]);
    let maxValueAdded = nums[0];
    
    for(let index = 1; index < nums.length; index++) {
        if(!mySet.has(nums[index])) {
            mySet.add(nums[index]);
            maxValueAdded = nums[index];
        } else {
            maxValueAdded++;
            mySet.add(maxValueAdded);
            result += maxValueAdded - nums[index];
        }
    }
    
    return result;
};